alienv

```diff
@@ -253,22 +257,44 @@ case "$ACTION" in
     MODULES=$(normModules "${ARGS[@]}")
     collectModules
     existModules $MODULES
-    eval $($MODULECMD bash add $MODULES 2> >(grep -v "$IGNORE_ERR" >&2))
+    PS1DEV=
+    if [[ $DEVOPT == 1 ]];then
+      PS1DEV=" (dev)"
+      for MODULE in $MODULES; do
+        source "$WORK_DIR/$ARCHITECTURE/$MODULE/etc/profile.d/init.sh"
+      done
+    else
+      eval $($MODULECMD bash add $MODULES 2> >(grep -v "$IGNORE_ERR" >&2))
+    fi
     [[ $UNAME == Darwin ]] && eval $(stripDyld)
     detectShell
     if [[ ! -z "$CLEAN_ENV" ]]; then
       case $MODULES_SHELL in
-        sh|bash) export PS1="[$MODULES]"' \w $> '   ;;
-        ksh)     export PS1="[$MODULES]"' $PWD $> ' ;;
-        zsh)     export PS1="[$MODULES]"' %~ %#> '  ;;
+        sh|bash) export PS1="[$MODULES]$PS1DEV"' \w $> '   ;;
+        ksh)     export PS1="[$MODULES]$PS1DEV"' $PWD $> ' ;;
+        zsh)     export PS1="[$MODULES]$PS1DEV"' %~ %#> '  ;;
       esac
     fi
     $MODULECMD bash list
-    printf "${ET}Use ${EM}alienv list${ET} to list loaded modules. " >&2
+    if [[ $DEVOPT == 1 ]];then
+      echo "*** Development env loaded for '$MODULES' ***" >&2
+    else
+      printf "${ET}Use ${EM}alienv list${ET} to list loaded modules. " >&2
+    fi
     printf "Use ${EM}exit${ET} to exit this environment.${EZ}\n" >&2
     exec ${CLEAN_ENV:+env $SHELL_NORC_ENV} $MODULES_SHELL ${CLEAN_ENV:+$SHELL_NORC_PARAM} -i
   ;;
   printenv|load|unload)
+    if [[ $DEVOPT == 1 ]];then
+      echo "*** Please run following commands to load dev environment for '$MODULES' ***" >&2
+      echo "" >&2
+      echo "export WORK_DIR=$WORK_DIR" >&2
+      for MODULE in $MODULES; do
+        echo "source $WORK_DIR/$ARCHITECTURE/$MODULE/etc/profile.d/init.sh" >&2
+      done
+      echo "" >&2
+      exit 0
+    fi
```

Comment:
This should definitely go after the verification and normalisation steps -- i.e. after the `MODULES=$(normModules ...)` and `collectModules` lines, so that your `$MODULES` variable has the correct format.

I see you've also changed the script so that *all* the lines go to stderr -- this again means that the following won't do what you might expect:

```bash
eval "$(alienv --dev load MyPackage/v1)"
```

...because it'll print the environment it gets from the modulefiles on stdout, which bash will eval, but not the development environment.

Ideally, this should probably be handled a bit like the `enter` case, where you *either* output `source .../init.sh` lines *or* call `modulecmd`, though you'd have to generate the correct syntax for each possible `$MODULES_SHELL`, keeping in mind that `source` is a bash- and zsh-specific extension (the portable POSIX syntax is just a plain dot, `.`).

If that's all a bit too complicated, I'd recommend just outputting a warning like you do for `setenv`. When the `$ACTION` is `unload`, you'd definitely need this warning, as I'm not sure resetting the environment is possible easily.